File: fixtures/python_demo/irisflow/io_layer.py
"""Reading and writing the edges of the pipeline."""
from __future__ import annotations

import pandas as pd

from .utils import log_step, require_columns

RAW_COLUMNS = [" Sepal.Length", "Sepal Width ", "Petal.Length", "PETAL WIDTH", "Species"]


def load_raw(path):
    """Load the messy source CSV."""
    df = pd.read_csv(path)
    require_columns(df, RAW_COLUMNS)
    return log_step("load_raw", df)


def save_result(df, path):
    """Persist the final frame."""
    df.to_csv(path, index=False)
    return log_step("save_result", df)

File: fixtures/r_demo/R/io_layer.R
# Reading and writing the edges of the pipeline.

suppressPackageStartupMessages({
  library(readr)
})

RAW_COLUMNS <- c("Sepal.Length", "Sepal Width", "Petal.Length", "PETAL WIDTH", "Species")

#' Load the messy source CSV.
load_raw <- function(path) {
  df <- read_csv(path, show_col_types = FALSE, name_repair = "minimal")
  names(df) <- trimws(names(df))
  require_columns(df, RAW_COLUMNS)
  log_step("load_raw", df)
}

#' Persist the final frame.
save_result <- function(df, path) {
  write_csv(df, path)
  log_step("save_result", df)
}
